### PR TITLE
Firmware testing

### PR DESCRIPTION
**Pull request by patrick-gu**
Right now it's just using JNI

- - -
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/or-airbrake-plugin/19)

### CODE REVIEW

**Comment by patrick-gu** [comment]
only tested on Windows with MSYS2

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# force off-tree build
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+  message(FATAL_ERROR "CMake generation is not allowed within the source directory!
+  Remove the CMakeCache.txt file and try again from another folder, e.g.:
+    mkdir build && cd build
+    cmake ..
+  ")
+endif()
+
+# default to Release build
```
Comment (celery6) [inline CMakeLists.txt:34]: idk if im reading this right, but proc uses Debug in production rn. Release is broken/unmaintained

**Review by celery6 — APPROVED** [review]

**Comment by celery6** [comment]
tested briefly, can confirm it uses the proc traj lib calculation

had to make several changes to make this build/run in my environment... it would be nice if anyone could easily use this but uh too bad i guess?

**Comment by patrick-gu** [comment]
I guess the latest changes haven't really been reviewed but I think they're minor enough that I can merge this

**Pull request merged**